vars/pr.groovy

```diff
@@ -0,0 +1,109 @@
+import groovy.json.JsonSlurperClassic
+
+//
+// Get PR's data
+//
+Map getData(Map config = [:]) {
+    // optional PR number or branch name
+    // if empty - use current branch
+    String pr = config.get('pr', '')
+    String branch = config.get('branch', '')
+    String url = config.get('url', '')
+    // optional `OWNER/REPO`
+    // if empty - use current repo or from `url`
+    String repo = config.get('repo', '')
+    // optional GH CLI credentials
+    String credentialsId = config.get('credentialsId', 'github-vega-ci-bot-artifacts')
+    // List of fields to get for a PR
+    List<String> prFields = config.get('prFields', ['body'])
+
+    String repoArgument = repo ? "--repo '${repo}'" : ''
+    String prOrBranchOrURL = pr ?: ( branch ?: url )
+
+    String cliResult = null
+
+    withGHCLI('credentialsId': credentialsId) {
+        cliResult = sh(
+            label: "Get data for a PR: ${prOrBranchOrURL}",
+            returnStdout: true,
+            script: "gh pr ${repoArgument} view ${prOrBranchOrURL} --json ${prFields.join(',')}"
+        ).trim()
+
+        echo "GH CLI result: ${cliResult}"
+    }
+
+    return new JsonSlurperClassic().parseText(cliResult)
+}
+
+String getFirstComment(Map config = [:]) {
+    Map prData = getData(config + ['prFields': ['body']])
+    return prData.body
+}
+
+List<String> getAllComments(Map config = [:]) {
+    List<String> result = []
+    Map prData = getData(config + ['prFields': ['body', 'comments']])
+    result += prData.body
+    prData.comments.each { commentData ->
+        result += commentData.body
+    }
+    return result
+}
+
+//
+// Parse First/Body comment of a PR to find
+//   information about connected changes in other repos
+//
+Map getConnectedChangesInOtherRepos(Map config = [:]) {
+    Map result = [:]
+    List<String> allComments = getAllComments(config)
+
+    Map repoToParam = [
+        'vega': 'VEGA_CORE_BRANCH',
+        'data-node': 'DATA_NODE_BRANCH',
+        'vegawallet': 'VEGAWALLET_BRANCH',
+        'ethereum-event-forwarder': 'ETHEREUM_EVENT_FORWARDER_BRANCH',
+        'devops-infra': 'DEVOPS_INFRA_BRANCH',
+        'vegatools': 'VEGATOOLS_BRANCH',
+        'system-tests': 'SYSTEM_TESTS_BRANCH',
+        'protos': 'PROTOS_BRANCH',
+        'networks': 'NETWORKS_BRANCH',
+        'checkpoint-store': 'CHECKPOINT_STORE_BRANCH'
```

Comment:
Doesn't look like changes often, but do we want "smart-contracts" in here also? Was just looking at the list of things I have checked out.

Reply:
`smart-contracts` repo is not used in CI (yet).